### PR TITLE
.Net: Clean up some prompt-related APIs

### PR DESCRIPTION
**Pull request by stephentoub**
Mostly XML comment cleanup / standardization. Some code cleanup (e.g. arg validation, avoiding unnecessary use of LINQ, etc.). The only public API changes were changing a few public consts to be public static properties and making a few properties/arguments nullable.

### CODE REVIEW

**Review by SergeyMenshykh — APPROVED** [review]

**Review by dmytrostruk — APPROVED** [review]

File: dotnet/src/InternalUtilities/src/Http/HttpHeaderValues.cs [hunk 1]
```diff
@@ -6,5 +6,5 @@ namespace Microsoft.SemanticKernel.Http;
 internal static class HttpHeaderValues
 {
     /// <summary>User agent string to use for all HTTP requests issued by Semantic Kernel.</summary>
-    public const string UserAgent = "Semantic-Kernel";
+    public static string UserAgent => "Semantic-Kernel";
```
[the remark this answers is not in the document]
Comment (RogerBarreto) [reply dotnet/src/InternalUtilities/src/Http/HttpHeaderValues.cs:9]: AFAIK for this scenario `const` is a better approach for this scenario as this is not bound or be defined at runtime.

**Review by markwallace-microsoft — APPROVED** [review]

File: dotnet/src/InternalUtilities/src/Http/HttpHeaderValues.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephentoub) [reply dotnet/src/InternalUtilities/src/Http/HttpHeaderValues.cs:9]: consts are baked into every use site. It effectively means you can never change their value, ever. We typically recommend const in public surface area only in situations it's required due to callers needing a const, eg the value needs to be used in an attribute.

[the remark this answers is not in the document]
Comment (WeihanLi) [reply dotnet/src/InternalUtilities/src/Http/HttpHeaderValues.cs:9]: It's `internal` and nowhere updates its value now, so think it does not make much sense updating to static property with getter only?

[the remark this answers is not in the document]
Comment (stephentoub) [reply dotnet/src/InternalUtilities/src/Http/HttpHeaderValues.cs:9]: There's zero benefit to this being a const, and having it be a const means it repeatedly shows up in searches when auditing for things to change.

**Pull request merged**